java/component/keyple-core/src/main/java/org/eclipse/keyple/core/seproxy/plugin/reader/AbstractLocalReader.java

```diff
@@ -897,22 +740,26 @@ private ApduResponse case4HackGetResponse(int originalStatusCode) {
   }
 
   /**
-   * Transmits a single APDU and receives its response.
+   * Transmits a single APDU and receives its response. Both are in the form of an array of bytes.
+   *
+   * <p>This abstract method must be implemented by the ProxyReader plugin (e.g. Pcsc, Nfc).
    *
-   * <p>This abstract method must be implemented by the ProxyReader plugin (e.g. Pcsc, Nfc). The
-   * implementation must handle the case where the SE response is 61xy and execute the appropriate
-   * get response command.
+   * <p><b>Caution: the implementation must handle the case where the SE response is 61xy and
+   * execute the appropriate get response command.</b>
    *
-   * @param apduIn byte buffer containing the ingoing data
+   * @param apduIn byte buffer containing the ingoing data (should be not null).
    * @return apduResponse byte buffer containing the outgoing data.
    * @throws KeypleReaderIOException if the communication with the reader or the SE has failed
+   * @since 0.9
    */
   protected abstract byte[] transmitApdu(byte[] apduIn);
 
   /**
    * Method to be implemented by child classes in order to handle the needed actions when
```

Comment:
It seems that this method could only be implemented by AbstractObservableLocalReader!
Is it correct to define an abstract method not implemented by child classes which aren't ObservableReader?

Reply:
See above comment about finalizeSeProcessing.

Reply:
terminateSeCommunication reverted